### ISSUE TITLE
v5.0.0+ breaks flake8-print plugin per-file-ignore settings

### ISSUE THREAD

**Issue by hans2520** [PyCQA/flake8#1663]
### how did you install flake8?

```console
pip install flake8
```

### unmodified output of `flake8 --bug-report`

```json
flake8 --bug-report
{
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.8.13",
    "system": "Linux"
  },
  "plugins": [
    {
      "plugin": "flake8-black",
      "version": "0.3.3"
    },
    {
      "plugin": "flake8-isort",
      "version": "4.2.0"
    },
    {
      "plugin": "flake8-print",
      "version": "5.0.0"
    },
    {
      "plugin": "mccabe",
      "version": "0.7.0"
    },
    {
      "plugin": "pycodestyle",
      "version": "2.9.1"
    },
    {
      "plugin": "pyflakes",
      "version": "2.5.0"
    }
  ],
  "version": "5.0.4"
}
```

### describe the problem

Files defined in per-file-ignores under flake8 settings should not throw errors for flake8-print formatting.

Pinning flake8 to v4.0.1 and everything works as expected. Adjusting the version of flake8-plugin has no effect.
...

#### sample code

```python
per-file-ignores =
  setup.py: T20
```

#### commands ran

```console
flake8 --output-file flake8.txt --exit-zero
flake8_junit flake8.txt flake8_junit.xml
flake8 --benchmark
...
```

**Comment by hans2520** [comment]
Note that leaving inline `#noqa T201` type comments does still work on the newer version of flake8

**Comment by asottile** [comment]
you're going to have to show more information like a file or what you expect or what you're not seeing -- there's not enough information in your report to understand what you're asking

**Comment by asottile** [comment]
works fine for me:

```console
$ flake8 t.py t2.py 
t2.py:1:1: T201 print found.
$ tail -n99 t.py t2.py setup.cfg 
==> t.py <==
print('hi')

==> t2.py <==
print('hi')

==> setup.cfg <==
[flake8]
per-file-ignores =
    t.py: T
```

**Comment by hans2520** [comment]
i.e. starting on flake8 v5.0.0+
```
File flake8_junit.xml was created successfullylint-ci run-test: commands[3] | flake8 --benchmark
[9101] /home/ubuntu/GitHub/insight$ /home/ubuntu/GitHub/insight/.tox/lint-ci/bin/flake8 --benchmark
./setup.py:14:5: T201 print found.
./setup.py:16:5: T201 print found.
./setup.py:20:5: T201 print found.
./setup.py:22:5: T201 print found.
```

When the per-file-ignore settings define that the `T20` error classes should be ignored for this file.

With flake8 v4.0.1 and previous, no errors are thrown

**Comment by hans2520** [comment]
@asottile The wrong button was accidentally clicked. Please re-open.

**Comment by asottile** [comment]
I will not reopen until you've put in the effort to make an actual reproducible example

**Comment by hans2520** [comment]
@asottile 

The [flake8] section with the per-file-ignore settings were in a tox.ini.

However, upon further investigation, there was another [flake8] section in a setup.cfg file also in the same repo.

The contents of this setup.cfg flake8 section was as follows:
```
[flake8]
max-line-length = 120
extend-ignore = D, E203, W503  # Inherit from black: https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html#flake8
```

This file was added for compatibility with black settings, as suggested in the link in the comment.

Removing the setup.cfg from the configuration, and the tests results on v5.0.4 pass as expected (respects per-file-ignore settings). Furthermore, putting the relevant settings that were present in setup.cfg and placing them into the tox.ini flake8 section, and the tests still produce expected results.

So it seems that there is a behavior change with v5.0.0+ of flake8, where, with multiple [flake8] sections across different config files, and different settings in each, the individualized settings of each section are respected in each file, except for the print plugin WRT to the `per-file-ignore` settings.

The full setup.cfg contents were provided above. The full original [flake8] section of the tox.ini are below:

```
application-import-names = myapp
ignore = W605,C901,E722,W503,W504
per-file-ignores =
  **/config/**: T20
  cli.py: T20
  setup.py: T20
exclude =
  .tox,
  .git,
  __pycache__,
  *.pyc,
  *.egg-info,
  .cache,
  .eggs,
max-complexity = 10
count = True
statistics = True
max-line-length = 120
```

Again, removing the setup.cfg (and placing the extend-ignore settings from it into the tox.ini) is a valid workaround

**Comment by asottile** [comment]
yeah you were in undefined territory before -- now flake8 only looks at files containing `[flake8]` (and stops at the first one) (the 5.0 changelog mentions this)

but none of your original issue came even close to describing your scenario -- in the future please actually follow the template and fill it out properly -- we don't have access to your computer so we can't possibly help you if you don't provide any information about your problem
